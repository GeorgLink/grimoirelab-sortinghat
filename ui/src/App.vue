<template>
  <v-app>
    <v-app-bar app color="primary" dark>
      <div class="d-flex align-center">
        <v-img
          alt="Vuetify Logo"
          class="shrink mr-2"
          contain
          src="https://cdn.vuetifyjs.com/images/logos/vuetify-logo-dark.png"
          transition="scale-transition"
          width="40"
        />

        <v-img
          alt="Vuetify Name"
          class="shrink mt-1 hidden-sm-and-down"
          contain
          min-width="100"
          src="https://cdn.vuetifyjs.com/images/logos/vuetify-name-dark.png"
          width="100"
        />
      </div>

      <v-spacer></v-spacer>

      <v-btn
        href="https://github.com/vuetifyjs/vuetify/releases/latest"
        target="_blank"
        text
      >
        <span class="mr-2">Latest Release</span>
        <v-icon>mdi-open-in-new</v-icon>
      </v-btn>
    </v-app-bar>

    <v-content>
      <work-space
        :highlight-individual="highlightInWorkspace"
        :individuals="savedIndividuals"
        :merge-items="mergeItems"
        :move-item="moveItem"
        @clearSpace="savedIndividuals = []"
        @updateIndividuals="updateTable"
        @highlight="highlightIndividual($event, 'highlightInTable', true)"
        @stopHighlight="highlightIndividual($event, 'highlightInTable', false)"
        @deselect="deselectIndividuals"
      />
      <v-row>
        <v-col class="individuals elevation-2">
          <individuals-table
            :fetch-page="getIndividualsPage"
            :delete-item="deleteItem"
            :merge-items="mergeItems"
            :unmerge-items="unmergeItems"
            :move-item="moveItem"
            :highlight-individual="highlightInTable"
            @saveIndividual="addSavedIndividual"
            @updateWorkspace="updateWorkspace"
            @highlight="
              highlightIndividual($event, 'highlightInWorkspace', true)
            "
            @stopHighlight="
              highlightIndividual($event, 'highlightInWorkspace', false)
            "
            ref="table"
          />
        </v-col>
        <v-col class="organizations elevation-2">
          <organizations-table
            :fetch-page="getOrganizationsPage"
            :enroll="enroll"
            :add-organization="addOrganization"
            :add-domain="addDomain"
            @updateIndividuals="updateTable"
            @updateWorkspace="updateWorkspace"
          />
        </v-col>
      </v-row>
      <v-snackbar v-model="snackbar">
        Individual already in work space
      </v-snackbar>
    </v-content>
  </v-app>
</template>

<script>
import {
  getPaginatedIndividuals,
  getPaginatedOrganizations
} from "./apollo/queries";
import {
  deleteIdentity,
  merge,
  unmerge,
  moveIdentity,
  enroll,
  addOrganization,
  addDomain
} from "./apollo/mutations";
import IndividualsTable from "./components/IndividualsTable";
import OrganizationsTable from "./components/OrganizationsTable";
import WorkSpace from "./components/WorkSpace";

export default {
  name: "App",
  components: {
    IndividualsTable,
    OrganizationsTable,
    WorkSpace
  },
  data() {
    return {
      highlightInTable: undefined,
      highlightInWorkspace: undefined,
      savedIndividuals: [],
      snackbar: false
    };
  },
  methods: {
    async getIndividualsPage(page, items) {
      const response = await getPaginatedIndividuals(this.$apollo, page, items);
      return response;
    },
    async getOrganizationsPage(page, items) {
      const response = await getPaginatedOrganizations(
        this.$apollo,
        page,
        items
      );
      return response;
    },
    addSavedIndividual(individual) {
      const isSaved = this.savedIndividuals.find(savedIndividual => {
        return individual.uuid === savedIndividual.uuid;
      });
      if (isSaved) {
        this.snackbar = true;
        return;
      }
      this.savedIndividuals.push(individual);
    },
    async deleteItem(uuid) {
      const response = await deleteIdentity(this.$apollo, uuid);
      return response;
    },
    async mergeItems(fromUuids, toUuid) {
      const response = await merge(this.$apollo, fromUuids, toUuid);
      return response;
    },
    updateTable() {
      this.$refs.table.queryIndividuals();
    },
    updateWorkspace(event) {
      if (event.remove) {
        event.remove.forEach(removedItem => {
          const removedIndex = this.savedIndividuals.findIndex(
            individual => individual.uuid == removedItem
          );
          if (removedIndex !== -1) {
            this.savedIndividuals.splice(removedIndex, 1);
          }
        });
      }
      if (event.update) {
        event.update.forEach(updatedItem => {
          const updatedIndex = this.savedIndividuals.findIndex(
            individual => individual.uuid == updatedItem.uuid
          );
          if (updatedIndex !== -1) {
            Object.assign(this.savedIndividuals[updatedIndex], updatedItem);
          }
        });
      }
    },
    highlightIndividual(individual, component, highlight) {
      this[component] = highlight ? individual.uuid : undefined;
    },
    async unmergeItems(uuids) {
      const response = await unmerge(this.$apollo, uuids);
      return response;
    },
    async moveItem(fromUuid, toUuid) {
      const response = await moveIdentity(this.$apollo, fromUuid, toUuid);
      return response;
    },
    deselectIndividuals() {
      this.$refs.table.deselectIndividuals();
    },
    async enroll(uuid, organization) {
      const response = await enroll(this.$apollo, uuid, organization);
      return response;
    },
    async addOrganization(organization) {
      const response = await addOrganization(this.$apollo, organization);
      return response;
    },
    async addDomain(domain, organization) {
      const response = await addDomain(this.$apollo, domain, organization);
      return response;
    }
  }
};
</script>
<style scoped>
.row {
  justify-content: space-between;
  margin: 32px;
}
.individuals {
  max-width: 1200px;
  width: 70%;
}
.organizations {
  width: 25%;
  max-width: 500px;
  align-self: flex-start;
  margin-left: 32px;
}
h4 {
  padding: 12px 26px;
}
</style>
