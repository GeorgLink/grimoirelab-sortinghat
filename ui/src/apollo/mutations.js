import gql from "graphql-tag";

const LOCK_INDIVIDUAL = gql`
  mutation LockIndividual($uuid: String!) {
    lock(uuid: $uuid) {
      uuid
      individual {
        isLocked
      }
    }
  }
`;

const UNLOCK_INDIVIDUAL = gql`
  mutation UnlockIndividual($uuid: String!) {
    unlock(uuid: $uuid) {
      uuid
      individual {
        isLocked
      }
    }
  }
`;

const DELETE_IDENTITY = gql`
  mutation DeleteIdentity($uuid: String!) {
    deleteIdentity(uuid: $uuid) {
      uuid
    }
  }
`;

const MERGE = gql`
  mutation Merge($fromUuids: [String!], $toUuid: String!) {
    merge(fromUuids: $fromUuids, toUuid: $toUuid) {
      uuid
      individual {
        isLocked
        identities {
          name
          source
          email
          uuid
          username
        }
        profile {
          name
          id
        }
        enrollments {
          organization {
            name
          }
        }
      }
    }
  }
`;

const UNMERGE = gql`
  mutation unmerge($uuids: [String!]) {
    unmergeIdentities(uuids: $uuids) {
      uuids
      individuals {
        profile {
          name
          id
          isBot
        }
        identities {
          name
          source
          email
          uuid
          username
        }
        enrollments {
          start
          end
          organization {
            name
          }
        }
      }
    }
  }
`;

const MOVE_IDENTITY = gql`
  mutation moveIdentity($fromUuid: String!, $toUuid: String!) {
    moveIdentity(fromUuid: $fromUuid, toUuid: $toUuid) {
      uuid
      individual {
        isLocked
        identities {
          name
          source
          email
          uuid
          username
        }
        profile {
          name
          id
        }
        enrollments {
          organization {
            name
          }
        }
      }
    }
  }
`;

const lockIndividual = (apollo, uuid) => {
  let response = apollo.mutate({
    mutation: LOCK_INDIVIDUAL,
    variables: {
      uuid: uuid
    }
  });
  return response;
};

const unlockIndividual = (apollo, uuid) => {
  let response = apollo.mutate({
    mutation: UNLOCK_INDIVIDUAL,
    variables: {
      uuid: uuid
    }
  });
  return response;
};

const deleteIdentity = (apollo, uuid) => {
  let response = apollo.mutate({
    mutation: DELETE_IDENTITY,
    variables: {
      uuid: uuid
    }
  });
  return response;
};

const merge = (apollo, fromUuids, toUuid) => {
  let response = apollo.mutate({
    mutation: MERGE,
    variables: {
      fromUuids: fromUuids,
      toUuid: toUuid
    }
  });
  return response;
};

const unmerge = (apollo, uuids) => {
  let response = apollo.mutate({
    mutation: UNMERGE,
    variables: {
      uuids: uuids
    }
  });
  return response;
};

const moveIdentity = (apollo, fromUuid, toUuid) => {
  let response = apollo.mutate({
    mutation: MOVE_IDENTITY,
    variables: {
      fromUuid: fromUuid,
      toUuid: toUuid
    }
  });
  return response;
};

export {
  lockIndividual,
  unlockIndividual,
  deleteIdentity,
  merge,
  unmerge,
  moveIdentity
};
