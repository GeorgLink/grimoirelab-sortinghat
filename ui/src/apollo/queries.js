import gql from "graphql-tag";

const GET_INDIVIDUAL_BYUUID = gql`
  query GetIndividual($uuid: String!) {
    individuals(filters: { uuid: $uuid }) {
      entities {
        mk
        identities {
          name
        }
        profile {
          id
          name
        }
      }
    }
  }
`;

const GET_INDIVIDUALS = gql`
  query GetIndividuals($page: Int!, $pageSize: Int!) {
    individuals(page: $page, pageSize: $pageSize) {
      entities {
        mk
        isLocked
        identities {
          name
          source
          uuid
        }
        profile {
          id
          name
        }
      }
      pageInfo {
        page
        pageSize
        numPages
        hasNext
        hasPrev
        startIndex
        endIndex
        totalResults
      }
    }
  }
`;

const GET_PROFILE_BYUUID = gql`
  query GetProfileByUuid($uuid: String!) {
    individuals(filters: { uuid: $uuid }) {
      entities {
        isLocked
        profile {
          name
        }
        identities {
          name
          source
          email
          uuid
          username
        }
        enrollments {
          start
          end
          organization {
            name
          }
        }
      }
    }
  }
`;

const getIndividualByUuid = (apollo, uuid) => {
  let response = apollo.query({
    query: GET_INDIVIDUAL_BYUUID,
    variables: {
      uuid: uuid
    }
  });
  return response;
};

const getIndividuals = {
  currentPage: 1,
  numPages: 1,
  pageSize: 10,
  query: async function(apollo, pageSize) {
    let self = this;
    if (pageSize) {
      self.pageSize = pageSize;
    }
    let response = await apollo.query({
      query: GET_INDIVIDUALS,
      variables: {
        page: self.currentPage,
        pageSize: pageSize
      }
    });
    self.numPages = response.data.individuals.pageInfo.numPages;
    if (self.currentPage > self.numPages) {
      return undefined;
    }
    self.currentPage++;
    return response;
  }
};

const getProfileByUuid = (apollo, uuid) => {
  let response = apollo.query({
    query: GET_PROFILE_BYUUID,
    variables: {
      uuid: uuid
    }
  });
  return response;
};

export { getIndividuals, getIndividualByUuid, getProfileByUuid };
