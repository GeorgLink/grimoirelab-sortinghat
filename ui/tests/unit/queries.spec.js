import { shallowMount } from "@vue/test-utils";
import Vue from "vue";
import Vuetify from "vuetify";
import App from "@/App";
import * as Queries from "@/apollo/queries";

Vue.use(Vuetify);

const responseMocked = {
  data: {
    individuals: {
      entities: [
        {
          mk: "172188fd88c1df2dd6d187b6f32cb6aced544aee",
          identities: [{ name: "test name", __typename: "IdentityType" }],
          profile: {
            id: "7",
            name: "test name",
            __typename: "ProfileType"
          },
          __typename: "IndividualType"
        }
      ],
      __typename: "IdentityPaginatedType"
    },
    pageInfo: {
      page: 1,
      pageSize: 10,
      numPages: 1,
      hasNext: false,
      hasPrev: false,
      startIndex: 1,
      endIndex: 1,
      totalResults: 1
    }
  }
};

describe("App", () => {
  test("mock query for getIndividuals", async () => {
    const query = jest.fn(() => Promise.resolve(responseMocked));
    const wrapper = shallowMount(App, {
      Vue,
      mocks: {
        $apollo: {
          query
        }
      }
    });
    let response = await Queries.getIndividuals(wrapper.vm.$apollo);
    let individuals_mocked = response.data;
    await wrapper.setData({
      individuals_mocked
    });

    expect(query).toBeCalled();
    expect(wrapper.element).toMatchSnapshot();
    expect(wrapper.vm.individuals).toBe(responseMocked.data.individuals);
  });

  test("getIndividuals with arguments", async () => {
    const getIndividualsSpied = spyOn(Queries, "getIndividuals");
    
    let response = await Queries.getIndividuals(undefined, 10, 100);
    expect(getIndividualsSpied).toHaveBeenLastCalledWith(undefined, 10, 100);
  });

  test("getIndividuals without arguments in the App component", async () => {
    const getIndividualsSpied = spyOn(Queries, "getIndividuals");
    const query = jest.fn(() => Promise.resolve(responseMocked));
    const wrapper = shallowMount(App, {
      Vue,
      mocks: {
        $apollo: {
          query
        }
      }
    });

    expect(getIndividualsSpied).toBeCalled();
    expect(getIndividualsSpied).toHaveBeenCalledWith(wrapper.vm.$apollo);
  });
});
