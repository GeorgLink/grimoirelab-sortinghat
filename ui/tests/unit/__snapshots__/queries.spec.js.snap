// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IndividualsData mock query for getIndividuals 1`] = `
<individualsgrid-stub
  class="grid"
  individuals="[object Object]"
/>
`;

exports[`IndividualsTable Mock query for getCountries 1`] = `
<v-container-stub
  tag="div"
>
  <v-row-stub
    class="actions"
    tag="div"
  >
    <h4
      class="title"
    >
      <v-icon-stub
        color="black"
        left=""
      >
        
        mdi-account-multiple
      
      </v-icon-stub>
      
      Individuals (0)
    
    </h4>
     
    <v-btn-stub
      activeclass=""
      class="black--text"
      color="secondary"
      depressed="true"
      tag="button"
      type="button"
    >
      
      Add
    
    </v-btn-stub>
  </v-row-stub>
   
  <v-row-stub
    class="actions"
    tag="div"
  >
    <v-checkbox-stub
      backgroundcolor=""
      errorcount="1"
      errormessages=""
      indeterminateicon="$checkboxIndeterminate"
      label="Select all"
      messages=""
      officon="$checkboxOff"
      onicon="$checkboxOn"
      ripple="true"
      rules=""
      successmessages=""
      value=""
      valuecomparator="function deepEqual(a, b) {
          if (a === b) return true;

          if (a instanceof Date && b instanceof Date && a.getTime() !== b.getTime()) {
            // If the values are Date, compare them as timestamps
            return false;
          }

          if (a !== Object(a) || b !== Object(b)) {
            // If the values aren't objects, they were already checked for equality
            return false;
          }

          var props = Object.keys(a);

          if (props.length !== Object.keys(b).length) {
            // Different number of props, don't bother to check
            return false;
          }

          return props.every(function (p) {
            return deepEqual(a[p], b[p]);
          });
        }"
    />
     
    <search-stub
      class="ma-0 ml-auto pa-0 flex-grow-0"
    />
     
    <v-tooltip-stub
      bottom="true"
      closedelay="0"
      contentclass=""
      fixed="true"
      maxwidth="auto"
      nudgebottom="0"
      nudgeleft="0"
      nudgeright="0"
      nudgetop="0"
      nudgewidth="0"
      opendelay="200"
      openonhover="true"
      tag="span"
      transition="expand-y-transition"
    >
       
      <span>
        Merge selected
      </span>
    </v-tooltip-stub>
     
    <v-tooltip-stub
      bottom="true"
      closedelay="0"
      contentclass=""
      fixed="true"
      maxwidth="auto"
      nudgebottom="0"
      nudgeleft="0"
      nudgeright="0"
      nudgetop="0"
      nudgewidth="0"
      opendelay="200"
      openonhover="true"
      tag="span"
      transition="expand-y-transition"
    >
       
      <span>
        Delete selected
      </span>
    </v-tooltip-stub>
  </v-row-stub>
   
  <v-data-table-stub
    customfilter="function defaultFilter(value, search, item) {
          return value != null && search != null && typeof value !== 'boolean' && value.toString().toLocaleLowerCase().indexOf(search.toLocaleLowerCase()) !== -1;
        }"
    customgroup="function groupItems(items, groupBy, groupDesc) {
          var key = groupBy[0];
          var groups = [];
          var current = null;

          for (var i = 0; i < items.length; i++) {
            var item = items[i];
            var val = getObjectValueByPath(item, key);

            if (current !== val) {
              current = val;
              groups.push({
                name: val,
                items: []
              });
            }

            groups[groups.length - 1].items.push(item);
          }

          return groups;
        }"
    customsort="function sortItems(items, sortBy, sortDesc, locale, customSorters) {
          if (sortBy === null || !sortBy.length) return items;
          var stringCollator = new Intl.Collator(locale, {
            sensitivity: 'accent',
            usage: 'sort'
          });
          return items.sort(function (a, b) {
            var _a, _b;

            for (var i = 0; i < sortBy.length; i++) {
              var sortKey = sortBy[i];
              var sortA = getObjectValueByPath(a, sortKey);
              var sortB = getObjectValueByPath(b, sortKey);

              if (sortDesc[i]) {
                _a = __read([sortB, sortA], 2), sortA = _a[0], sortB = _a[1];
              }

              if (customSorters && customSorters[sortKey]) {
                var customResult = customSorters[sortKey](sortA, sortB);
                if (!customResult) continue;
                return customResult;
              } // Check if both cannot be evaluated


              if (sortA === null && sortB === null) {
                continue;
              }

              _b = __read([sortA, sortB].map(function (s) {
                return (s || '').toString().toLocaleLowerCase();
              }), 2), sortA = _b[0], sortB = _b[1];

              if (sortA !== sortB) {
                if (!isNaN(sortA) && !isNaN(sortB)) return Number(sortA) - Number(sortB);
                return stringCollator.compare(sortA, sortB);
              }
            }

            return 0;
          });
        }"
    expanded=""
    expandicon="$expand"
    groupby=""
    groupdesc=""
    headers="[object Object],[object Object],[object Object],[object Object]"
    hidedefaultfooter="true"
    hidedefaultheader="true"
    itemclass=""
    itemkey="uuid"
    items=""
    itemsperpage="10"
    loaderheight="4"
    loadingtext="$vuetify.dataIterator.loadingText"
    locale="en-US"
    mobilebreakpoint="600"
    nodatatext="$vuetify.noDataText"
    noresultstext="$vuetify.dataIterator.noResultsText"
    options="[object Object]"
    page="0"
    selectablekey="isSelectable"
    serveritemslength="-1"
    sortby=""
    sortdesc=""
    value=""
  />
   
  <div
    class="text-center pt-2"
  >
    <v-pagination-stub
      currentpagearialabel="$vuetify.pagination.ariaLabel.currentPage"
      length="0"
      nextarialabel="$vuetify.pagination.ariaLabel.next"
      nexticon="$next"
      pagearialabel="$vuetify.pagination.ariaLabel.page"
      previcon="$prev"
      previousarialabel="$vuetify.pagination.ariaLabel.previous"
      totalvisible="7"
      value="0"
      wrapperarialabel="$vuetify.pagination.ariaLabel.wrapper"
    />
  </div>
   
  <v-dialog-stub
    closedelay="0"
    contentclass=""
    maxwidth="500px"
    opendelay="0"
    origin="center center"
    retainfocus="true"
    transition="dialog-transition"
    width="auto"
  >
    <v-card-stub
      loaderheight="4"
      tag="div"
    >
      <v-card-title-stub
        class="headline"
      >
        
      </v-card-title-stub>
       
      <v-card-text-stub>
        
      </v-card-text-stub>
       
      <v-card-actions-stub>
        <v-spacer-stub />
         
        <v-btn-stub
          activeclass=""
          tag="button"
          text="true"
          type="button"
        >
          
          Cancel
        
        </v-btn-stub>
         
        <v-btn-stub
          activeclass=""
          color="blue darken-4"
          tag="button"
          text="true"
          type="button"
        >
          
          Confirm
        
        </v-btn-stub>
      </v-card-actions-stub>
    </v-card-stub>
  </v-dialog-stub>
   
  <profile-modal-stub
    addidentity="() => {}"
    enroll="() => {}"
    getcountries="function mockConstructor() {
        return fn.apply(this, arguments);
      }"
    updateprofile="() => {}"
  />
   
  <v-card-stub
    class="dragged-item"
    color="primary"
    dark="true"
    loaderheight="4"
    tag="div"
  >
    <v-card-subtitle-stub>
      
      Moving
      0
      individual
    
    </v-card-subtitle-stub>
  </v-card-stub>
   
  <v-snackbar-stub
    contentclass=""
    tag="div"
    timeout="5000"
    transition="v-snack-transition"
  >
    
    
  
  </v-snackbar-stub>
</v-container-stub>
`;

exports[`IndividualsTable Mock query for getPaginatedIndividuals 1`] = `
<v-container-stub
  tag="div"
>
  <v-row-stub
    class="actions"
    tag="div"
  >
    <h4
      class="title"
    >
      <v-icon-stub
        color="black"
        left=""
      >
        
        mdi-account-multiple
      
      </v-icon-stub>
      
      Individuals (0)
    
    </h4>
     
    <v-btn-stub
      activeclass=""
      class="black--text"
      color="secondary"
      depressed="true"
      tag="button"
      type="button"
    >
      
      Add
    
    </v-btn-stub>
  </v-row-stub>
   
  <v-row-stub
    class="actions"
    tag="div"
  >
    <v-checkbox-stub
      backgroundcolor=""
      errorcount="1"
      errormessages=""
      indeterminateicon="$checkboxIndeterminate"
      label="Select all"
      messages=""
      officon="$checkboxOff"
      onicon="$checkboxOn"
      ripple="true"
      rules=""
      successmessages=""
      value=""
      valuecomparator="function deepEqual(a, b) {
          if (a === b) return true;

          if (a instanceof Date && b instanceof Date && a.getTime() !== b.getTime()) {
            // If the values are Date, compare them as timestamps
            return false;
          }

          if (a !== Object(a) || b !== Object(b)) {
            // If the values aren't objects, they were already checked for equality
            return false;
          }

          var props = Object.keys(a);

          if (props.length !== Object.keys(b).length) {
            // Different number of props, don't bother to check
            return false;
          }

          return props.every(function (p) {
            return deepEqual(a[p], b[p]);
          });
        }"
    />
     
    <search-stub
      class="ma-0 ml-auto pa-0 flex-grow-0"
    />
     
    <v-tooltip-stub
      bottom="true"
      closedelay="0"
      contentclass=""
      fixed="true"
      maxwidth="auto"
      nudgebottom="0"
      nudgeleft="0"
      nudgeright="0"
      nudgetop="0"
      nudgewidth="0"
      opendelay="200"
      openonhover="true"
      tag="span"
      transition="expand-y-transition"
    >
       
      <span>
        Merge selected
      </span>
    </v-tooltip-stub>
     
    <v-tooltip-stub
      bottom="true"
      closedelay="0"
      contentclass=""
      fixed="true"
      maxwidth="auto"
      nudgebottom="0"
      nudgeleft="0"
      nudgeright="0"
      nudgetop="0"
      nudgewidth="0"
      opendelay="200"
      openonhover="true"
      tag="span"
      transition="expand-y-transition"
    >
       
      <span>
        Delete selected
      </span>
    </v-tooltip-stub>
  </v-row-stub>
   
  <v-data-table-stub
    customfilter="function defaultFilter(value, search, item) {
          return value != null && search != null && typeof value !== 'boolean' && value.toString().toLocaleLowerCase().indexOf(search.toLocaleLowerCase()) !== -1;
        }"
    customgroup="function groupItems(items, groupBy, groupDesc) {
          var key = groupBy[0];
          var groups = [];
          var current = null;

          for (var i = 0; i < items.length; i++) {
            var item = items[i];
            var val = getObjectValueByPath(item, key);

            if (current !== val) {
              current = val;
              groups.push({
                name: val,
                items: []
              });
            }

            groups[groups.length - 1].items.push(item);
          }

          return groups;
        }"
    customsort="function sortItems(items, sortBy, sortDesc, locale, customSorters) {
          if (sortBy === null || !sortBy.length) return items;
          var stringCollator = new Intl.Collator(locale, {
            sensitivity: 'accent',
            usage: 'sort'
          });
          return items.sort(function (a, b) {
            var _a, _b;

            for (var i = 0; i < sortBy.length; i++) {
              var sortKey = sortBy[i];
              var sortA = getObjectValueByPath(a, sortKey);
              var sortB = getObjectValueByPath(b, sortKey);

              if (sortDesc[i]) {
                _a = __read([sortB, sortA], 2), sortA = _a[0], sortB = _a[1];
              }

              if (customSorters && customSorters[sortKey]) {
                var customResult = customSorters[sortKey](sortA, sortB);
                if (!customResult) continue;
                return customResult;
              } // Check if both cannot be evaluated


              if (sortA === null && sortB === null) {
                continue;
              }

              _b = __read([sortA, sortB].map(function (s) {
                return (s || '').toString().toLocaleLowerCase();
              }), 2), sortA = _b[0], sortB = _b[1];

              if (sortA !== sortB) {
                if (!isNaN(sortA) && !isNaN(sortB)) return Number(sortA) - Number(sortB);
                return stringCollator.compare(sortA, sortB);
              }
            }

            return 0;
          });
        }"
    expanded=""
    expandicon="$expand"
    groupby=""
    groupdesc=""
    headers="[object Object],[object Object],[object Object],[object Object]"
    hidedefaultfooter="true"
    hidedefaultheader="true"
    itemclass=""
    itemkey="uuid"
    items=""
    itemsperpage="10"
    loaderheight="4"
    loadingtext="$vuetify.dataIterator.loadingText"
    locale="en-US"
    mobilebreakpoint="600"
    nodatatext="$vuetify.noDataText"
    noresultstext="$vuetify.dataIterator.noResultsText"
    options="[object Object]"
    page="0"
    selectablekey="isSelectable"
    serveritemslength="-1"
    sortby=""
    sortdesc=""
    value=""
  />
   
  <div
    class="text-center pt-2"
  >
    <v-pagination-stub
      currentpagearialabel="$vuetify.pagination.ariaLabel.currentPage"
      length="0"
      nextarialabel="$vuetify.pagination.ariaLabel.next"
      nexticon="$next"
      pagearialabel="$vuetify.pagination.ariaLabel.page"
      previcon="$prev"
      previousarialabel="$vuetify.pagination.ariaLabel.previous"
      totalvisible="7"
      value="0"
      wrapperarialabel="$vuetify.pagination.ariaLabel.wrapper"
    />
  </div>
   
  <v-dialog-stub
    closedelay="0"
    contentclass=""
    maxwidth="500px"
    opendelay="0"
    origin="center center"
    retainfocus="true"
    transition="dialog-transition"
    width="auto"
  >
    <v-card-stub
      loaderheight="4"
      tag="div"
    >
      <v-card-title-stub
        class="headline"
      >
        
      </v-card-title-stub>
       
      <v-card-text-stub>
        
      </v-card-text-stub>
       
      <v-card-actions-stub>
        <v-spacer-stub />
         
        <v-btn-stub
          activeclass=""
          tag="button"
          text="true"
          type="button"
        >
          
          Cancel
        
        </v-btn-stub>
         
        <v-btn-stub
          activeclass=""
          color="blue darken-4"
          tag="button"
          text="true"
          type="button"
        >
          
          Confirm
        
        </v-btn-stub>
      </v-card-actions-stub>
    </v-card-stub>
  </v-dialog-stub>
   
  <profile-modal-stub
    addidentity="() => {}"
    enroll="() => {}"
    getcountries="() => {}"
    updateprofile="() => {}"
  />
   
  <v-card-stub
    class="dragged-item"
    color="primary"
    dark="true"
    loaderheight="4"
    tag="div"
  >
    <v-card-subtitle-stub>
      
      Moving
      0
      individual
    
    </v-card-subtitle-stub>
  </v-card-stub>
   
  <v-snackbar-stub
    contentclass=""
    tag="div"
    timeout="5000"
    transition="v-snack-transition"
  >
    
    
  
  </v-snackbar-stub>
</v-container-stub>
`;

exports[`JobsTable Mock query for getJobs 1`] = `
<v-container-stub
  tag="div"
>
  <h4
    class="title mt-5 mb-5"
  >
    <v-icon-stub
      color="black"
      left=""
    >
      
      mdi-tray-full
    
    </v-icon-stub>
    
    Jobs
  
  </h4>
   
  <!---->
   
  <p
    class="text-subtitle-1"
  >
    There are no jobs in the queue.
  </p>
</v-container-stub>
`;

exports[`OrganizationsTable Mock query for getPaginatedOrganizations 1`] = `
<v-container-stub
  tag="div"
>
  <v-row-stub
    class="actions"
    tag="div"
  >
    <h4
      class="title"
    >
      <v-icon-stub
        color="black"
        left=""
      >
        
        mdi-sitemap
      
      </v-icon-stub>
      
      Organizations (0)
    
    </h4>
     
    <v-btn-stub
      activeclass=""
      class="black--text"
      color="secondary"
      depressed="true"
      tag="button"
      type="button"
    >
      
      Add
    
    </v-btn-stub>
  </v-row-stub>
   
  <v-row-stub
    class="actions"
    tag="div"
  >
    <search-stub
      class="ma-0 mr-3 pa-0 flex-grow-0"
    />
  </v-row-stub>
   
  <v-data-table-stub
    customfilter="function defaultFilter(value, search, item) {
          return value != null && search != null && typeof value !== 'boolean' && value.toString().toLocaleLowerCase().indexOf(search.toLocaleLowerCase()) !== -1;
        }"
    customgroup="function groupItems(items, groupBy, groupDesc) {
          var key = groupBy[0];
          var groups = [];
          var current = null;

          for (var i = 0; i < items.length; i++) {
            var item = items[i];
            var val = getObjectValueByPath(item, key);

            if (current !== val) {
              current = val;
              groups.push({
                name: val,
                items: []
              });
            }

            groups[groups.length - 1].items.push(item);
          }

          return groups;
        }"
    customsort="function sortItems(items, sortBy, sortDesc, locale, customSorters) {
          if (sortBy === null || !sortBy.length) return items;
          var stringCollator = new Intl.Collator(locale, {
            sensitivity: 'accent',
            usage: 'sort'
          });
          return items.sort(function (a, b) {
            var _a, _b;

            for (var i = 0; i < sortBy.length; i++) {
              var sortKey = sortBy[i];
              var sortA = getObjectValueByPath(a, sortKey);
              var sortB = getObjectValueByPath(b, sortKey);

              if (sortDesc[i]) {
                _a = __read([sortB, sortA], 2), sortA = _a[0], sortB = _a[1];
              }

              if (customSorters && customSorters[sortKey]) {
                var customResult = customSorters[sortKey](sortA, sortB);
                if (!customResult) continue;
                return customResult;
              } // Check if both cannot be evaluated


              if (sortA === null && sortB === null) {
                continue;
              }

              _b = __read([sortA, sortB].map(function (s) {
                return (s || '').toString().toLocaleLowerCase();
              }), 2), sortA = _b[0], sortB = _b[1];

              if (sortA !== sortB) {
                if (!isNaN(sortA) && !isNaN(sortB)) return Number(sortA) - Number(sortB);
                return stringCollator.compare(sortA, sortB);
              }
            }

            return 0;
          });
        }"
    expanded=""
    expandicon="$expand"
    groupby=""
    groupdesc=""
    headers="[object Object],[object Object],[object Object]"
    hidedefaultfooter="true"
    hidedefaultheader="true"
    itemclass=""
    itemkey="id"
    items=""
    itemsperpage="10"
    loaderheight="4"
    loadingtext="$vuetify.dataIterator.loadingText"
    locale="en-US"
    mobilebreakpoint="600"
    nodatatext="$vuetify.noDataText"
    noresultstext="$vuetify.dataIterator.noResultsText"
    options="[object Object]"
    page="0"
    selectablekey="isSelectable"
    serveritemslength="-1"
    sortby=""
    sortdesc=""
    value=""
  />
   
  <div
    class="text-center pt-2"
  >
    <v-pagination-stub
      currentpagearialabel="$vuetify.pagination.ariaLabel.currentPage"
      length="0"
      nextarialabel="$vuetify.pagination.ariaLabel.next"
      nexticon="$next"
      pagearialabel="$vuetify.pagination.ariaLabel.page"
      previcon="$prev"
      previousarialabel="$vuetify.pagination.ariaLabel.previous"
      totalvisible="5"
      value="0"
      wrapperarialabel="$vuetify.pagination.ariaLabel.wrapper"
    />
  </div>
   
  <organization-modal-stub
    adddomain="() => {}"
    addorganization="() => {}"
    deletedomain="() => {}"
    domains=""
  />
   
  <v-dialog-stub
    closedelay="0"
    contentclass=""
    maxwidth="500px"
    opendelay="0"
    origin="center center"
    retainfocus="true"
    transition="dialog-transition"
    width="auto"
  >
    <v-card-stub
      class="pa-3"
      loaderheight="4"
      tag="div"
    >
      <v-card-title-stub
        class="headline"
      >
        
      </v-card-title-stub>
       
      <v-card-text-stub>
        
      </v-card-text-stub>
       
      <v-card-actions-stub>
        <v-spacer-stub />
         
        <v-btn-stub
          activeclass=""
          tag="button"
          text="true"
          type="button"
        >
          
          Cancel
        
        </v-btn-stub>
         
        <v-btn-stub
          activeclass=""
          color="primary"
          depressed="true"
          tag="button"
          type="button"
        >
          
          Confirm
        
        </v-btn-stub>
      </v-card-actions-stub>
    </v-card-stub>
  </v-dialog-stub>
   
  <v-snackbar-stub
    contentclass=""
    tag="div"
    timeout="5000"
    transition="v-snack-transition"
  >
    
    
  
  </v-snackbar-stub>
   
  <v-card-stub
    class="dragged-organization"
    color="primary"
    dark="true"
    loaderheight="4"
    tag="div"
  >
    <v-card-title-stub>
      
      
    
    </v-card-title-stub>
     
    <v-card-subtitle-stub>
      
      Drag and drop on an individual to affiliate
    
    </v-card-subtitle-stub>
  </v-card-stub>
</v-container-stub>
`;
